x/application/keeper/msg_server_undelegate_from_gateway_test.go

```diff
@@ -332,24 +324,40 @@ func TestMsgServer_UndelegateFromGateway_DelegationIsActiveUntilNextSession(t *t
 	require.NotNil(t, app)
 
 	// Verify that when queried for the next session the reconstructed delegatee
-	// gateway list does include the undelegated gateway.
+	// gateway list does not include the undelegated gateway.
 	nextSessionGatewayAddresses := rings.GetRingAddressesAtBlock(&app, nextSessionStartHeight)
-	require.NotContains(t, nextSessionGatewayAddresses, gatewayAddrToUndelegate)
+	require.NotContains(t, nextSessionGatewayAddresses, pendingUndelegateToAddr)
+
+	// Increment the block height past the tested session's grace period and run
+	// the pruning undelegations logic again.
+	sessionGracePeriodBlockCount := uint64(sessionkeeper.GetSessionGracePeriodBlockCount())
+	afterSessionGracePeriodHeight := int64(sessionEndHeight + sessionGracePeriodBlockCount + 1)
+	sdkCtx = sdkCtx.WithBlockHeight(afterSessionGracePeriodHeight)
+	k.EndBlockerPruneAppToGatewayPendingUndelegation(sdkCtx)
+
+	// Verify that when queried for a block height past the tested session's grace period,
+	// the reconstructed delegatee gateway list does not include the undelegated gateway.
+	pastGracePeriodGatewayAddresses := rings.GetRingAddressesAtBlock(&app, afterSessionGracePeriodHeight)
+	require.NotContains(t, pastGracePeriodGatewayAddresses, pendingUndelegateToAddr)
+
+	// Verify that when queried for a block height corresponding to the past session
+	// that has its grace period elapsed, the reconstructed delegatee gateway list
```

Comment:
`that has its grace period elapsed` - I don't fully understand this. What am I missing about the explanation?

Reply:
```suggestion
	// for which the grace period has elapsed, the reconstructed delegatee gateway list
```